flytekit/extras/tensorflow/model.py

```diff
@@ -0,0 +1,78 @@
+import pathlib
+from typing import Type
+
+import tensorflow as tf
+
+from flytekit.core.context_manager import FlyteContext
+from flytekit.core.type_engine import TypeEngine, TypeTransformer, TypeTransformerFailedError
+from flytekit.models.core import types as _core_types
+from flytekit.models.literals import Blob, BlobMetadata, Literal, Scalar
+from flytekit.models.types import LiteralType
+
+
+class TensorflowModelTransformer(TypeTransformer[tf.keras.Model]):
+    TENSORFLOW_FORMAT = "TensorflowModel"
+
+    def __init__(
+        self,
+    ):
+        super().__init__(name="Tensorflow Model", t=tf.keras.Model)
+
+    def get_literal_type(self, t: Type[tf.keras.Model]) -> LiteralType:
+        return LiteralType(
+            blob=_core_types.BlobType(
+                format=self.TENSORFLOW_FORMAT,
+                dimensionality=_core_types.BlobType.BlobDimensionality.SINGLE,
+            )
+        )
+
+    def to_literal(
+        self,
+        ctx: FlyteContext,
+        python_val: tf.keras.Model,
+        python_type: Type[tf.keras.Model],
+        expected: LiteralType,
+    ) -> Literal:
+        meta = BlobMetadata(
+            type=_core_types.BlobType(
+                format=self.TENSORFLOW_FORMAT,
+                dimensionality=_core_types.BlobType.BlobDimensionality.SINGLE,
+            )
+        )
+
+        local_path = ctx.file_access.get_random_local_path() + ".h5"
```

Comment:
```suggestion
        local_path = ctx.file_access.get_random_local_path()
```

Reply:
`tf.keras.models.save_model` supports [2 formats](https://www.tensorflow.org/guide/keras/save_and_serialize#whole-model_saving_loading), by default it is `SavedModel` format but it stores the model as a folder and not as a single file. Due to which this line fails
```
ctx.file_access.put_data(local_path, remote_path, is_multipart=False)
```
So I used the `h5` format since it stores it as a single file.

Reply:
You should be able to copy a directory as well by setting `is_multipart` to True.